presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -177,7 +177,12 @@ public void finish()
     @Override
     public boolean isFinished()
     {
-        return finishing && aggregationBuilder == null && (outputIterator == null || !outputIterator.hasNext());
+        boolean finished = finishing && aggregationBuilder == null && (outputIterator == null || !outputIterator.hasNext());
+        if (finished) {
+            // prevent reporting the last aggregationBuilder's usage to the memoryPool
+            operatorContext.setMemoryReservation(0);
```

Comment:
Is this necessary? When `Driver.destroyIfNecessary()` is called it will release all memory from all operator contexts